packages/vector_graphics_codec/lib/vector_graphics_codec.dart

```diff
@@ -0,0 +1,631 @@
+    buffer._putInt32(paintId);
+    return paintId;
+  }
+
+  void _readFillPaint(_ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final int color = buffer.getUint32();
+    final int blendMode = buffer.getUint8();
+    final int id = buffer.getInt32();
+
+    listener?.onPaintObject(
+      color,
+      null,
+      null,
+      blendMode,
+      null,
+      null,
+      0, // Fill
+      id,
+    );
+  }
+
+  void _readStrokePaint(_ReadBuffer buffer, VectorGraphicsCodecListener? listener) {
+    final int color = buffer.getUint32();
+    final int strokeCap = buffer.getUint8();
+    final int strokeJoin = buffer.getUint8();
+    final int blendMode = buffer.getUint8();
+    final double strokeMiterLimit = buffer.getFloat64();
+    final double strokeWidth = buffer.getFloat64();
+    final int id = buffer.getInt32();
+
+    listener?.onPaintObject(
+      color,
+      strokeCap,
+      strokeJoin,
+      blendMode,
+      strokeMiterLimit,
+      strokeWidth,
+      1, // Stroke
+      id,
+    );
+  }
+
+  /// Begin writing a new path to the [buffer], returing the identifier
+  /// assigned to it.
+  ///
+  /// The [fillType] argument is either `1` for a fill or `0` for a stroke.
+  ///
+  /// Throws a [StateError] if there is already an active path.
+  int writeStartPath(VectorGraphicsBuffer buffer, int fillType) {
+    if (buffer._currentPathId != -1) {
+      throw StateError('There is already an active Path.');
+    }
+    if (buffer._decodePhase.index > _DecodePhase.paths.index) {
+      throw StateError('Paths must be encoded together');
+    }
+    buffer._decodePhase = _DecodePhase.paths;
+    buffer._currentPathId = buffer._nextPathId++;
+    buffer._putUint8(_pathTag);
+    buffer._putUint8(fillType);
+    buffer._putInt32(buffer._currentPathId);
+    return buffer._currentPathId;
+  }
+
+  /// Write a move to command to the global coordinate ([x], [y]).
+  ///
+  /// Throws a [StateError] if there is not an active path.
+  void writeMoveTo(VectorGraphicsBuffer buffer, double x, double y) {
+    if (buffer._currentPathId == -1) {
+      throw StateError('There is no active Path.');
+    }
+    buffer._putUint8(_moveToTag);
+    buffer._putFloat64(x);
+    buffer._putFloat64(y);
+  }
+
+  /// Write a line to command to the global coordinate ([x], [y]).
+  ///
+  /// Throws a [StateError] if there is not an active path.
+  void writeLineTo(VectorGraphicsBuffer buffer, double x, double y) {
+    if (buffer._currentPathId == -1) {
+      throw StateError('There is no active Path.');
+    }
+    buffer._putUint8(_lineToTag);
+    buffer._putFloat64(x);
+    buffer._putFloat64(y);
+  }
+
+  /// Write an arc to command to the global coordinate ([x1], [y1]) with control
+  /// points CP1 ([x2], [y2]) and CP2 ([x3], [y3]).
+  ///
+  /// Throws a [StateError] if there is not an active path.
+  void writeArcTo(VectorGraphicsBuffer buffer, double x1, double y1, double x2, double y2, double x3, double y3) {
```

Comment:
I think this is supposed to be `writeCubicTo`. arcToPoint takes different parameters than this.

Reply:
Done